src/frontend/flight/handler.rs

```diff
@@ -89,4 +104,121 @@ impl SeafowlFlightHandler {
+        let log_store = match cmd.store {
+            None => self.context.internal_object_store.get_log_store(&cmd.path),
+            Some(store_loc) => self
+                .context
+                .metastore
+                .object_stores
+                .get_log_store_for_table(
+                    Url::parse(&store_loc.location).map_err(|e| {
+                        DataFusionError::Execution(format!(
+                            "Couldn't parse sync location: {e}"
+                        ))
+                    })?,
+                    store_loc.options,
+                    cmd.path,
+                )?,
+        };
+
+        // Check if a table exists yet in the provided location, and if not create one.
+        // If it does fetch its durable sequence number.
+        let dur_seq = if !log_store.is_delta_table_location().await? {
+            // Get the actual table schema by removing the `SEAFOWL_SYNC_DATA_UD_FLAG` column
+            let schema = batches.first().unwrap().schema();
+            let idxs = (0..schema.all_fields().len() - 1).collect::<Vec<usize>>();
+            let schema = schema.project(&idxs)?;
+
+            let delta_schema = DeltaSchema::try_from(&schema)?;
+
+            debug!(
+                "Creating new Delta table at location: {}",
+                log_store.root_uri()
+            );
+            CreateBuilder::new()
+                .with_log_store(log_store.clone())
+                .with_columns(delta_schema.fields().clone())
+                .with_comment(format!("Created by Seafowl {}", env!("CARGO_PKG_VERSION")))
+                .await?;
+            None
+        } else {
+            let mut table = DeltaTable::new(log_store.clone(), Default::default());
+            table.load().await?;
+
+            // TODO: handle all edge cases with missing/un-parsable sequence numbers
+            let commit_infos = table.history(Some(1)).await?;
+            match commit_infos
+                .last()
+                .expect("Table has non-zero commits")
+                .info
+                .get(SEAFOWL_SYNC_DATA_SEQUENCE_NUMBER)
+            {
+                Some(Value::Number(seq)) => seq.as_u64(),
+                _ => None,
+            }
+        };
+
+        // Get the current memory sequence number, defaulting to durable sequence number if missing
+        let url = log_store.root_uri();
+        let mem_seq = self
+            .sync_manager
+            .read()
+            .await
+            .mem_seq_for_table(&url)
+            .or(dur_seq);
+
+        let num_rows = batches
+            .iter()
+            .fold(0, |rows, batch| rows + batch.num_rows());
+        if num_rows == 0 {
+            debug!("Received empty batches, returning current sequence numbers");
+            return Ok(DataSyncResult {
+                accepted: true,
+                memory_sequence_number: mem_seq,
+                durable_sequence_number: dur_seq,
+            });
+        }
```

Comment:
Ah something like "if this hasn't been invoked during the lifetime of `SeafowlSyncManager` return this in all cases and set some flag, otherwise return only if the flag was set **and** num_rows == 0"?

Reply:
When a new client first connects (basically, we want to make sure that if Seafowl restarts and the client reconnects, the client knows which offset to restart from, so it sends an empty DoPut without any data to get the startup LSNs, and, as a precaution, Seafowl would reject DoPuts that aren't empty).